recipes/ezc3d/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "ezc3d-Release_{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        # don't force PIC
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "set(CMAKE_POSITION_INDEPENDENT_CODE ON)", "")
+        # fix install
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "set(${PROJECT_NAME}_LIB_FOLDER Lib)",
+                              "set(${PROJECT_NAME}_LIB_FOLDER lib)")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "set(${PROJECT_NAME}_LIB_FOLDER lib/${PROJECT_NAME})",
+                              "set(${PROJECT_NAME}_LIB_FOLDER lib)")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "set(${PROJECT_NAME}_BIN_FOLDER lib/${PROJECT_NAME})",
+                              "set(${PROJECT_NAME}_BIN_FOLDER bin)")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["USE_MATRIX_FAST_ACCESSOR"] = True
+        self._cmake.definitions["BINDER_PYTHON3"] = False
+        self._cmake.definitions["BINDER_MATLAB"] = False
+        self._cmake.definitions["BUILD_EXAMPLE"] = False
+        self._cmake.definitions["BUILD_DOC"] = False
+        self._cmake.definitions["GET_OFFICIAL_DOCUMENTATION"] = False
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        # FIXME: remove namespace for imported target
+        self.cpp_info.includedirs = [os.path.join("include", "ezc3d")]
```

Comment:
```suggestion
        self.cpp_info.includedirs = [os.path.join("include", "ezc3d")]
```
```suggestion
        self.cpp_info.includedirs.append(os.path.join("include", "ezc3d"))
```

Reply:
It's useless to add `include` dir, it's not -unfortunately- the official way to include those headers according to upstream.
For example, if os.path.join("include", "ezc3d") is removed, and you include `ezc3d/ezc3d_all.h` it fails because some others headers in subfolders will try to include others headers without "ezc3d/" dir

Reply:
It looks like ezc3d ships headers with regular names such as `Point.h`, `Data.h`...
See https://github.com/pyomeca/ezc3d/tree/dev/include
It would be nice to include these using `ezc3/Point.h` instead of `Point.h`.

Reply:
I know, I've opened an issue https://github.com/pyomeca/ezc3d/issues/165

But i don't think that conan should change interface of a lib. It should come from upstream.

Reply:
I think that issue is invalid, as the headers are already installed in a subdirectory (which is namespaced).
The issue is valid because it expects users to include the header using `#include "ezc3d_all.h"` instead of `#include "ezc3d/ezc3d_all.h"`.

And indeed, it is not up to us to rename these headers.
But adding the `include` directory to `self.cpp_info.includedirs` is not changing interface (by my suggestion).
When you would install it to your root at `/usr/include`, it would also be possible to include the headers using `#include "ezc3d/ezc3d_all.h"`

Reply:
It should work yes (because `ezc3d` is also added to include paths, otherwise it would fail), but for example "include" folder is not added to official CMake config file.

Reply:
When you're going for 100% compatibility using the targets, then you're correct.

I searched on github for includes of `ezc3d.h` and they all include it without using a subdirectory.

Reply:
Hey guys! :)
As discussed in the issue https://github.com/pyomeca/ezc3d/pull/166, I tried to add the interface `<path_to_ezc3d>/..` so #include <ezc3d/Point.h> works
Does this do the trick?